lib/node_modules/@stdlib/ndarray/some-by/lib/assign.js

```diff
@@ -0,0 +1,205 @@
+*
+* // Create an output ndarray:
+* var y = empty( [], {
+*     'dtype': 'bool'
+* });
+*
+* // Perform reduction:
+* var out = assign( x, 3, y, isEven );
+* // returns <ndarray>
+*
+* var v = out.get();
+* // returns true
+*/
+function assign( x, n, y, options, predicate, thisArg ) {
+	var nargs;
+	var opts;
+	var err;
+	var ord;
+	var ctx;
+	var cb;
+	var N;
+	var v;
+
+	if ( !isndarrayLike( x ) ) {
+		throw new TypeError( format( 'invalid argument. First argument must be an ndarray-like object. Value: `%s`.', x ) );
+	}
+	if ( !isndarrayLike( y ) ) {
+		throw new TypeError( format( 'invalid argument. Third argument must be an ndarray-like object. Value: `%s`.', y ) );
+	}
+
+	N = ndims( x );
+	nargs = arguments.length;
+
+	if ( nargs === 4 ) {
+		if ( !isFunction( options ) ) {
+			throw new TypeError( format( 'invalid argument. Predicate must be a function. Value: `%s`.', options ) );
+		}
+		cb = options;
+		ctx = thisArg;
+		opts = objectAssign( {}, defaults );
+		opts.dims = zeroTo( N );
+	} else if ( nargs === 5 ) {
+		if ( isFunction( options ) ) {
+			cb = options;
+			ctx = predicate;
+			opts = objectAssign( {}, defaults );
+			opts.dims = zeroTo( N );
+		}
+		if ( isObject( options ) ) {
+			if ( !isFunction( predicate ) ) {
+				throw new TypeError( format( 'invalid argument. Predicate must be a function. Value: `%s`.', predicate ) );
+			}
+			cb = predicate;
+			ctx = thisArg;
+			opts = objectAssign( {}, defaults, options );
+			err = validate( opts, N, options );
+			if ( err ) {
+				throw err;
+			}
+			if ( opts.dims === null ) {
+				opts.dims = zeroTo( N );
+			}
+		}
+	} else if ( nargs === 6 ) {
+		if ( !isObject( options ) ) {
+			throw new TypeError( format( 'invalid argument. Options argument must be an object. Value: `%s`.', options ) );
+		}
+		if ( !isFunction( predicate ) ) {
+			throw new TypeError( format( 'invalid argument. Predicate must be a function. Value: `%s`.', predicate ) );
+		}
+		cb = predicate;
+		ctx = thisArg;
+		opts = objectAssign( {}, defaults, options );
+		err = validate( opts, N, options );
+		if ( err ) {
+			throw err;
+		}
+		if ( opts.dims === null ) {
+			opts.dims = zeroTo( N );
+		}
+	}
+
+	// Resolve input array meta data:
+	ord = getOrder( x );
+
+	if ( isndarrayLike( n ) ) {
+		if ( !contains( TYPES, getDtype( n ) ) ) {
+			throw new TypeError( format( 'invalid argument. Second argument must be an ndarray-like object having an integer data type. Value: `%s`.', n ) );
+		}
+		try {
+			v = maybeBroadcastArray( n, getShape( y ) );
+		} catch ( err ) { // eslint-disable-line no-unused-vars
+			throw new Error( 'invalid argument. Second argument must be broadcast-compatible with the non-reduced dimensions of the input array.' );
+		}
+	} else {
+		if ( !isInteger( n ) ) {
+			throw new TypeError( format( 'invalid argument. Second argument must be an integer or an ndarray-like object. Value: `%s`.', n ) );
+		}
+		v = broadcastScalar( n, 'generic', getShape( y ), ord );
```

Comment:
Do not use `generic` here. If you use `generic`, you preclude us from being able to call into C into the future, as "generic" arrays cannot be easily supported in native code. Use the default `integer_index` dtype. See `ndarray/count-if` for how to resolve that dtype.